.github/actions/dev-env-setup/action.yml

```diff
@@ -0,0 +1,40 @@
+name: "Set up EED dev environment"
+description: "Sets up asdf, perl, and configures the cache"
+runs:
+  using: composite
+  steps:
+    - name: asdf setup
+      uses: asdf-vm/actions/setup@v1
+    - uses: shogo82148/actions-setup-perl@v1
+    - name: install pg perl library
+      run: sudo apt-get install -y libpq-dev libdbd-pg-perl
+      shell: bash
+    - name: set perl env variables
+      shell: bash
+      run: | # pragma: allowlist secret
+        echo "PERL5LIB=/home/runner/perl5/lib/perl5" >> $GITHUB_ENV
+        echo "PERL_LOCAL_LIB_ROOT=/home/runner/perl5" >> $GITHUB_ENV
+        echo "PERL_MB_OPT=--install_base '/home/runner/perl5'" >> $GITHUB_ENV
+        echo "PERL_MM_OPT=INSTALL_BASE=/home/runner/perl5" >> $GITHUB_ENV
+        echo "/home/runner/perl5/bin" >> $GITHUB_PATH
+        echo "CURRENT_PERL_VERSION=$(perl -e 'print "$^V\n"')" >> $GITHUB_ENV
+    - uses: actions/cache@v3
+      id: asdf-cache
+      with:
+        path: |
+          ~/.asdf
+        key: ${{ runner.os }}-asdf-cache-${{ hashFiles('.tool-versions') }}-v2
```

Comment:
you don't need those `v2`, they were added to invalidate some bad caches. Also we can delete caches with the github API now, so this shouldn't be needed ever again